vortex-array/src/array/varbin/mod.rs

```diff
@@ -73,20 +72,32 @@ impl VarBinArray {
             bytes_len: bytes.len(),
         };
 
-        let mut children = Vec::with_capacity(3);
-        children.push(offsets);
-        children.push(bytes);
-        if let Some(a) = validity.into_array() {
-            children.push(a)
-        }
+        let children = match validity.into_array() {
```

Comment:
there's a cute version here that we do in couple places `validity.into_array().into_iter().collect()`

Reply:
oh but you have offsets, ignore me